fastify.js

```diff
@@ -103,6 +103,9 @@ function fastify (options) {
     throw new FST_ERR_OPTIONS_NOT_OBJ()
   }
 
+  // Shallow copy options object to prevent mutations outside of this function
+  options = Object.assign({}, options)
```

Comment:
I would improve this line too

https://github.com/fastify/fastify/blob/7ddced69c45c294450ab12fe7f92f0d969fe86d7/fastify.js#L100

It can be moved here at 107 and become:

```js
options = options ? { ...options } : {}
```

Reply:
I had first considered this modification but I will skip check on line 102 `typeof options !== 'object'` and will get unexpected behavior when user uses invalid options such as `string`.
But maybe I'm getting the wrong idea ?

Reply:
fyi: I used [structuredClone() ](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone) as a workaround while this PR is finalized.

Reply:
I don't think structuredClone handle class?
```js
function Instance() {
  this.hello = 'world'
  let hidden = ''
  this.setHidden = (some) => {
    hidden = some
  }
  this.getHidden = () => hidden
}
var a = { instance: new Instance() }
var b = structuredClone(a) // Error
```

Reply:
> But maybe I'm getting the wrong idea ?


```js
function fastify (options) {
  // Options validations
  if (options && typeof options !== 'object') {
    throw new FST_ERR_OPTIONS_NOT_OBJ()
  } else {
    // Shallow copy options object to prevent mutations outside of this function
    options = Object.assign({}, options)
  }

  if (options.querystringParser && typeof options.querystringParser !== 'function') {
 ....
```